forge/db/controllers/Device.js

```diff
@@ -14,16 +14,17 @@ module.exports = {
                 device.set('activeSnapshotId', null)
             }
         } else {
-            // Check the snapshot is one we recognise
+            // Update the activeSnapshotId if valid and not already set
             const snapshotId = app.db.models.ProjectSnapshot.decodeHashid(state.snapshot)
             // hashid.decode returns an array of values, not the raw value.
-            if (snapshotId?.length > 0) {
-                // check to see if snapshot still exists
-                if (await app.db.models.ProjectSnapshot.byId(state.snapshot)) {
-                    device.set('activeSnapshotId', snapshotId)
+            if (snapshotId?.length > 0 && snapshotId !== device.activeSnapshotId) {
+                // Check to see if snapshot exists
+                if (await app.db.models.ProjectSnapshot.count({ where: { id: snapshotId }, limit: 1 }) > 0) {
+                    device.set('activeSnapshotId', snapshotId[0])
```

Comment:
One thing of note, I'm surprised this was ever working, previously it was being called with: `device.set('activeSnapshotId', [1])`, which would be trying to set the ID to an array.

Reply:
Sequelize has always appeared to do the right thing when passes an array when expecting a simple value. I've looked at this a couple times in the past and convinced myself all was okay.